### DISCUSSION TITLE
Does "Tunnel creation success rate" play any role in router/peer ranking in i2pd?

### DISCUSSION THREAD

**Discussion by bjtftw** [PurpleI2P/i2pd#1969] [General]
Why am I asking?

Because you can easily manipulate it with exploratory tunnels length setting.

For short exploratory tunnels setting (1 hop) I get like 80% "Tunnel creation success rate" while long tunnels (8 hops) lower this to 20-30% and standard length (3 hops) gives 50-60%.

So if that number (successful rate) play any important role in any decisions that i2pd/i2p make (like peer selection for new tunnels, anti-flood/fake routers ban list, etc) it may be not too objective about peer connectivity quality (assuming high % successful tunnel creation rate means good peer and low means bad peer).

So is a bug or a feature or something to rethink?

**Comment by Vort** [comment]
If I remember correctly, TCSR is used only as indicator for user and not broadcasted to the network.
For profiling similar approach is applied, but it measures how often transit tunnel creation fails.
While TCSR is calculated only for user tunnels and not for transit tunnels.

**Reply by bjtftw** [reply]
Ah OK, I was thinking that this number is published in router info (or something) and other peers do make decisions regarding it. So it's only my personal statistic.

**Comment by gh338** [comment]
My guess is that the profiling in i2pd is to 'blame' here. Naturaly the longer the tunnel the lower the rate will be. As the tunnel is 'serial' a failed peer in the tunnel will cause the tunnel to fail entirely. So the more tunnels you spin up for a destination (default for non exploratory is 5) the higher the rate should be (not sure if this is accurate regarding the code).

cheers

**Reply by Vort** [reply]
I agree for the first part (longer tunnels are harder to create).
But why do you think tunnel amount may affect creation rate?
